storage/src/system/auction.rs

```diff
@@ -100,11 +102,21 @@ pub trait Auction:
             }
             validator_bid.increase_stake(amount)?;
             validator_bid.with_delegation_rate(delegation_rate);
+            validator_bid.set_delegation_amount_boundaries(
+                minimum_delegation_amount,
+                maximum_delegation_amount,
+            );
```

Comment:
The way I think this should work is, if these settings are changed the forced undelegation logic should be called here to process out / undelegate any delegations above the max or below the min.

Reply:
That was our first idea, but during the initial discussion about implementation you suggested it'd be better to handle it in bulk at the end of an era before rewards are distributed.

If you think it's OK I'd gladly refactor it to handle forced undelegation within `add_bid`. Since we create standard unbonds it should not impact rewards distribution after all.

Reply:
There are pros and cons. If i recall correctly, I had a concern that a validator with a lot of delegators making a change would result in an unexpected long operation to loop over all of their delegators in the midst of executing a normal transaction in a normal block, and pushing it to the end as part of step processing where post processing is already expected is less surprising behavior. 

Now that the call to forced undelegate is present in step, I'm ok w/ proceeding with it being handled in that way.